File: solutions/ruby/difference-of-squares/1/difference_of_squares.rb
module BookKeeping
  VERSION = 3
end

class Squares
  def initialize(size)
    @size = size
  end

  def difference
    square_of_sum - sum_of_squares
  end

  def square_of_sum
    if @size <= 0
      return 0
    end

    (1..@size).inject(:+)**2
  end

  def sum_of_squares
    if @size <= 0
      return 0
    end
    
    (1..@size).map do |value|
      value**2
    end.inject(:+)
  end
end

File: solutions/fortran/difference-of-squares/1/difference_of_squares.f90
module difference_of_squares
   implicit none
contains
   integer function sum_of_squares(n)
      integer, intent(in) :: n
      sum_of_squares = n*(n + 1)*(2*n + 1)/6
   end function sum_of_squares
   integer function square_of_sum(n)
      integer, intent(in) :: n
      integer :: s
      s = n*(n + 1)/2
      square_of_sum = s*s
   end function square_of_sum
   integer function difference(n)
      integer, intent(in) :: n
      difference = square_of_sum(n) - sum_of_squares(n)
   end function difference
end module difference_of_squares